app/uk/gov/hmrc/homeofficesettledstatusstubs/controllers/NinoController.scala

```diff
@@ -22,70 +22,60 @@ import play.api.{Configuration, Environment}
 import uk.gov.hmrc.domain.Nino
 import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
 import uk.gov.hmrc.homeofficesettledstatusstubs.models._
+import uk.gov.hmrc.homeofficesettledstatusstubs.models.searches.NinoSearch
 import uk.gov.hmrc.homeofficesettledstatusstubs.services.StubDataService
 
 import java.util.UUID
 import javax.inject.{Inject, Singleton}
 import scala.concurrent.Future
 
 @Singleton
-class HomeOfficeSettledStatusStubsController @Inject()(
+class NinoController @Inject()(
   val env: Environment,
   cc: ControllerComponents,
   stubDataService: StubDataService)(implicit val configuration: Configuration)
     extends BackendController(cc) {
 
+  //todo add filter/action to add this.
   final val HTTP_HEADER_CONTENT_TYPE_JSON = "Content-Type" -> "application/json"
 
-  def token: Action[AnyContent] = Action.async { implicit request =>
-    Future.successful(
-      HomeOfficeSettledStatusStubsController.tokenForm
-        .bindFromRequest()
-        .fold(
-          errors => BadRequest(Json.toJson(errors.errors.map(e => (e.key, e.message)).toMap)),
-          _ => Ok(Json.parse(s"""{
-                                |   "access_token": "${UUID.randomUUID().toString}",
-                                |   "refresh_token": "${UUID.randomUUID().toString}",
-                                |   "token_type": "Bearer"
-                                |}""".stripMargin))
-        ))
-  }
-
-  def publicFundsByNino: Action[JsValue] = Action.async(parse.tolerantJson) { implicit request =>
+  def publicFundsByNino: Action[JsValue] = Action(parse.tolerantJson) { implicit request =>
     val correlationId = request.headers.get("X-Correlation-Id").getOrElse("00000000")
 
-    val response = withValidParameters(correlationId, request.body) {
-      (nino, givenName, familyName, dateOfBirth) =>
-        stubDataService.ninoSearch(correlationId, nino, givenName, familyName, dateOfBirth)
-    }
-
-    Future.successful(response.withHeaders(HTTP_HEADER_CONTENT_TYPE_JSON))
+    withValidParameters(correlationId, request.body)
+      .fold(
+        identity,
+        search => stubDataService.ninoSearch(search)
+      )
+      .withHeaders(HTTP_HEADER_CONTENT_TYPE_JSON)
   }
 
-  private def withValidParameters(correlationId: String, body: JsValue)(
-    successFunc: (String, String, String, String) => Result): Result = {
-
+  private def withValidParameters(
+    correlationId: String,
+    body: JsValue): Either[Result, NinoSearch] = {
     val ninoOpt = (body \ "nino").asOpt[String].map(_.replaceAll(" ", "").toUpperCase)
     val givenNameOpt = (body \ "givenName").asOpt[String]
     val familyNameOpt = (body \ "familyName").asOpt[String]
     val dateOfBirthOpt = (body \ "dateOfBirth").asOpt[String]
+    //todo start date end date
```

Comment:
Do we not just want to add these in here now then since we're introducing the logic for the other endpoint?

Reply:
will do I was weary that, it might start failing for existing, however thinking about it the front end always defaults right so should be fine 🤞

Reply:
:+1: